column_computation.hh

```diff
@@ -116,6 +119,9 @@ public:
     virtual column_computation_ptr clone() const override {
         return std::make_unique<collection_column_computation>(*this);
     }
+    virtual bool depends_on_non_primary_key_column() const override {
```

Comment:
Would be nice to copy the description of this function from the commit message to a comment here.
But I have a question about the description: _"detect a case where the primary key of a materialized view depends on a non primary key column from the base table in a view, but at the same time, the view itself doesn't have non-primary key columns."_
Is the last part of this description correct? I.e., it's not just the computation on a non-primary-key column, but **also** that this column is not in the view? If so, the comment is doubly-important, because the name of the function doesn't even hint this is the case (and also, it's not clear if "primary key" refers to a key in the base, or a key in the view).

Reply:
The last part is correct. I'm just stressing where this is an issue. But it doesn't mean that all the column computations with `depends_on_non_primary_key_column()==true` have the property that they don't contain the column itself.

I added slightly reworded comment to the source.

re `it's not clear if "primary key" refers to a key in the base, or a key in the view`: again, if the column computation is for a column in a view, it is referring to the primary key of the view. I don't think it is needed to further clarify that, as in principle, code in column_computation.hh is not inherently index-aware.